src/create-rpc-server.ts

```diff
@@ -0,0 +1,74 @@
+import { createServer } from 'node:http'
+import { writeHeapSnapshot } from 'node:v8'
+
+export interface RpcServerOptions {
+  apiPort?: number
+}
+
+export async function createRpcServer ({ apiPort }: RpcServerOptions): Promise<void> {
+  const rpcUsername = process.env.RPC_USERNAME
+  const rpcPassword = process.env.RPC_PASSWORD
+
+  if (rpcUsername == null || rpcPassword == null) {
+    // eslint-disable-next-line no-console
+    console.info('RPC_USERNAME and RPC_PASSWORD are not set. Not starting the RPC server.')
```

Comment:
imo it should be allowed to not set basic auth

It may be that the rpc is just exposed on a private network, so auth is happening at the network level.